src/torchcodec/decoders/_simple_video_decoder.py

```diff
@@ -186,17 +187,17 @@ def _getitem_slice(self, key: slice) -> Tensor:
         )
         return frame_data
 
-    def __getitem__(self, key: Union[int, slice]) -> Tensor:
+    def __getitem__(self, key: Union[numbers.Integral, slice]) -> Tensor:
         """Return frame or frames as tensors, at the given index or range.
 
         Args:
-            key(int or slice): The index or range of frame(s) to retrieve.
+            key(numbers.Integral or slice): The index or range of frame(s) to retrieve.
 
         Returns:
             torch.Tensor: The frame or frames at the given index or range.
         """
-        if isinstance(key, int):
-            return self._getitem_int(key)
+        if isinstance(key, numbers.Integral):
```

Comment:
One question: would it be more general to try to cast to an int and seeing if we get an exception instead of explicitly checking numbers.Integral here?

My concern is that the user may pass in a type that's castable to an int but not an instance of numbers.Integral (is that even possible?). Do you know if that is the case?

Reply:
Casting to an int would _succeed_ if the user provided a float or a string (for example, `decoder[2.3]` or `decoder["3"]`), but we want to fail in those cases. We don't want to check if an object can be castable to an int, because `int(2.3)` and `int("3")` will both succeed. We want to have the same behavior as Python lists, and both `lst[2.3]` and `lst["3"]` result in a TypeError. Rather, we want to check if the object has a conversion to int, which is what numbers.Integral does.

The other option is `if int(key) == key`, but I think that's a little too subtle. I'd rather do this, which I think is the blessed, modern way of checking for exactly what we want.

To answer your question: yes, I believe that is possible. But I think this handles the majority of cases we want to handle.